src/app/shared/components/template/services/instance/template-process.service.ts

```diff
@@ -21,38 +14,19 @@ import { TemplateFieldService } from "./template-field.service";
  * component extending it (instead of vice-versa)
  */
 @Injectable({ providedIn: "root" })
-export class TemplateProcessService {
+export class TemplateProcessService extends TemplateInstanceService {
   container: TemplateContainerComponent;
-  constructor(
-    templateService: TemplateService,
-    templateVariables: TemplateVariablesService,
-    templateTranslateService: TemplateTranslateService,
-    templateFieldService: TemplateFieldService,
-    tourService: TourService,
-    router: Router,
-    route: ActivatedRoute,
-    // elRef: ElementRef,
-    templateNavService: TemplateNavService,
-    // cdr: ChangeDetectorRef,
-    settingsService: SettingsService,
-    serverService: ServerService,
-    analyticsService: AnalyticsService
-  ) {
+  templateService: TemplateService;
+  templateNavService: TemplateNavService;
+  constructor(injector: Injector) {
+    super(injector);
+    this.templateService = this.getGlobalService(TemplateService);
+    this.templateNavService = this.getGlobalService(TemplateNavService);
     // Create mock template container component
```

Comment:
Further examples of the much-reduced amount of boilerplate required to create template processors and template containers